src/Enzo/enzo_EnzoInitialBCenter.cpp

```diff
@@ -0,0 +1,258 @@
+/// See LICENSE_CELLO file for license and copyright information
+/// @file     enzo_EnzoInitialBCenter.hpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     Wed Jul 3 2019
+/// @brief    [\ref Problem] VLCT Bfield initialization method implementation
+
+#include "enzo.hpp"
+
+//======================================================================
+
+EnzoInitialBCenter::EnzoInitialBCenter
+(Parameters * parameters,int cycle, double time, bool update_etot) throw ()
+  : Initial(cycle,time),
+    parameters_(parameters),
+    values_(),
+    update_etot_(update_etot)
+{
+  initialize_values_();
+}
+
+//----------------------------------------------------------------------
+
+void EnzoInitialBCenter::initialize_bfield_center( Block * block )
+{
+  // Simply sets the values of the cell-centered B-fields based on the
+  // previously initialized face-centered B-fields
+  EnzoConstrainedTransport ct;
+  Grouping bfieldc_group, bfieldi_group;
+  bfieldc_group.add("bfield_x","bfield");
+  bfieldc_group.add("bfield_y","bfield");
+  bfieldc_group.add("bfield_z","bfield");
+
+  bfieldi_group.add("bfieldi_x", "bfield");
+  bfieldi_group.add("bfieldi_y", "bfield");
+  bfieldi_group.add("bfieldi_z", "bfield");
+  ct.compute_center_bfield(block, 0, bfieldc_group, bfieldi_group);
+  ct.compute_center_bfield(block, 1, bfieldc_group, bfieldi_group);
+  ct.compute_center_bfield(block, 2, bfieldc_group, bfieldi_group);
+}
+
+//----------------------------------------------------------------------
+
+void bfieldi_helper_(EFlt3DArray &bfield,
+                     CelloArray<double,3> &Aj,
```

Comment:
Throughout the vast majority of the codebase I use `EFlt3DArray` which is an alias of `CelloArray<enzo_float,3>`

I was just using `CelloArray<double,3>` within some of the problem initializers. I believe that my rationale was to initialize problems as consistently as possible (so that the effects would be minimized when `enzo_float` changes). But since the expectation is that the code will compiled with associative floating point operations, I generally don't have particularly strong feelings about this.

I think in the existing initializers the use of `double` appears to be the convention (at least this seems to be the the case for `EnzoInitialCollapse`, `EnzoInitialSoup`, `InitialValue`). I think this may be partially informed by the fact that the floating point values from the configuration file are always parsed and stored in double precision.

In this particular case, I think it makes particular sense to use `double` values because this class supports initializing magnetic fields from Vector Potentials specified in the parameter file with arbitrary formulas. This mirrors the functionality provided in the `InitialValue` class (which also uses `double` precision for initialization).